Doc/glossary.rst

```diff
@@ -226,6 +226,12 @@ Glossary
       A variable defined in a class and intended to be modified only at
       class level (i.e., not in an instance of the class).
 
+   closure variable
+      A variable referenced from a :term:`nested scope` that is defined in an outer scope
+      rather than being resolved at runtime from the globals or builtin namespaces.
+      May be explicitly defined with the :keyword:`nonlocal` keyword to allow write access,
+      or implicitly defined if the variable is only read without being written to.
```

Comment:
Both `a` and `b` are closure *and* free variables in `mid` (due to lexical scoping, every free variable in `inner` is also a free variable in `mid`, even if `mid` doesn't otherwise reference the name).

In the example, `print` is a free variable without being a closure variable, though.

```suggestion
      A :term:`free variable` referenced from a :term:`nested scope` that is defined in an outer
      scope rather than being resolved at runtime from the globals or builtin namespaces.
      May be explicitly defined with the :keyword:`nonlocal` keyword to allow write access,
      or implicitly defined if the variable is only read without being written to.
      
      For example, in the ``inner`` function in the following code, both ``x`` and ``print`` are
      :term:`free variables <free variable>`, but only ``x`` is a *closure variable*::
      
          def outer():
              x = 0
              def inner():
                  nonlocal x
                  x += 1
                  print(x)
              return inner
```